src/Type/Php/JsonValidateTypeSpecifyingExtension.php

```diff
@@ -0,0 +1,42 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Analyser\SpecifiedTypes;
+use PHPStan\Analyser\TypeSpecifier;
+use PHPStan\Analyser\TypeSpecifierAwareExtension;
+use PHPStan\Analyser\TypeSpecifierContext;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\Accessory\AccessoryNonFalsyStringType;
+use PHPStan\Type\FunctionTypeSpecifyingExtension;
+use function count;
+
+class JsonValidateTypeSpecifyingExtension implements FunctionTypeSpecifyingExtension, TypeSpecifierAwareExtension
```

Comment:
I tried using a stub signature instead, but this did not work for me locally.

as I can see this extension also only work in PHP 8.3.. I will retry using a stub signature on a different composer which has PHP 8.3 installed

Reply:
ok, did a few more tests. It does not work with pure stub signatures, because these would also affect the ELSE case in a wrong way:

```
1) PHPStan\Analyser\NodeScopeResolverTest::testFileAsserts with data set "/Users/staabm/workspace/phpstan-src/tests/PHPStan/Analyser/data/json_validate.php:13" ('type', '/Users/staabm/workspace/phpst...te.php', PHPStan\Type\Constant\ConstantStringType Object (...), PHPStan\Type\MixedType Object (...), 13)
Expected type mixed, got type mixed~non-falsy-string in /Users/staabm/workspace/phpstan-src/tests/PHPStan/Analyser/data/json_validate.php on line 13.
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'mixed'
+'mixed~non-falsy-string'
```